pkg/kube/pod_runner.go

```diff
@@ -66,3 +71,26 @@ func (p *podRunner) Run(ctx context.Context, fn func(context.Context, *v1.Pod) (
 	}()
 	return fn(ctx, pod)
 }
+
+// RunEx will create a new Pod based on PodRunner contents and execute the given function
+func (p *podRunner) RunEx(ctx context.Context, fn PodRunnerFunc) (PodOutputMap, error) {
```

Comment:
It looks like this function is mostly copy-paste for Run. Can we reduce code duplication?

Reply:
the problem with the Run is that even though it already uses a PodController, it still calls a functor with the pod's name, and the functor works with the pod directly, not through the pod controller.
the RunEx was an attempt to move all users of the function step by step to use under the controller.

Reply:
But I still see that they share 99% of each other. Can we reduce code duplication?

Reply:
Done.